pettingzoo/utils/conversions.py

```diff
@@ -260,8 +260,8 @@ def add_new_agent(self, new_agent):
     def step(self, action):
         if self.dones[self.agent_selection]:
             del self._actions[self.agent_selection]
-            return self._was_done_step(action)
-        self._actions[self.agent_selection] = action
+            return self._was_done_step(action)  # type: ignore
+        self._actions[self.agent_selection] = action  # type: ignore
```

Comment:
Not sure how to remove those 2 `# type: ignore` statements, would anyone have any suggestion?

Reply:
So I just checked, `_was_done_step` actually returns `None`.

Reply:
@jjshoots I tried your suggestion and I still get the same typing error, but maybe I missed something? :thinking:

Reply:
Do you need to do `-> None` at the function definition?

Reply:
Could you perhaps point me to where it is? Thanks!

Reply:
Here is is: https://github.com/Farama-Foundation/PettingZoo/pull/737/commits/511df7479770cd28ef148e8a935c31fb9ca84052.

Reply:
I think you could just do the same thing as earlier. That whole function returns None, not sure why it was written that way instead of 
```py
xxx()
return
```

Reply:
I tried it but it didn't work. Now I put `def step(self, action: None):` and it seems the type checker is happy with that but I honestly don't understand it as `action` is sometimes `int` or `None`... :thinking:

Reply:
I'm not too familiar with this, but does `Optional[int]` not work?

Reply:
`Optional[int]` is just a synonym for `Union[int, None]`, i.e. it can be either an int or None

Reply:
Yes but kir0ul mentioned that it was sometimes `int` or `None`, since the typechecker is happy with it being `None`, a more semantically accurate type hint would be `Optional[int]`, no?

Reply:
I tried `Optional[int]` too but I get:
```
./pettingzoo/utils/conversions.py
  ./pettingzoo/utils/conversions.py:264:40 - error: Argument of type "int | None" cannot be assigned to parameter "action" of type "None" in function "_was_done_step"
    Type "int | None" cannot be assigned to type "None"
      Type cannot be assigned to type "None" (reportGeneralTypeIssues)
  ./pettingzoo/utils/conversions.py:265:9 - error: Argument of type "int | None" cannot be assigned to parameter "__v" of type "None" in function "__setitem__"
    Type "int | None" cannot be assigned to type "None"
      Type cannot be assigned to type "None" (reportGeneralTypeIssues)
```
Also sometimes the type of `action` is `np.int32`.

Reply:
This seems like a really annoying case of previously not enforcing types. Is it possible for us to fix the `np.int32` thing using a type cast.

Reply:
Done in https://github.com/Farama-Foundation/PettingZoo/pull/737/commits/55c69e687298090dd8a19a23051e042929fe65ca#diff-3ef3437e8b392d934e60182d2116f18ce516d3685c43a4b40c2a5f84c0dfe93fR267.